packages/features/bookings/lib/getBookingResponsesSchema.ts

```diff
@@ -76,8 +95,16 @@ function preprocess<T extends z.ZodType>({
           : z.string().refine((val) => isValidPhoneNumber(val));
         // Tag the message with the input name so that the message can be shown at appropriate place
         const m = (message: string) => `{${bookingField.name}}${message}`;
+        const views = bookingField.views;
+        const isFieldApplicableToCurrentView =
+          currentView === "ALL_VIEWS" ? true : views ? views.find((view) => view.id === currentView) : true;
         // If the field is hidden, then it can never be required
-        const isRequired = bookingField.hidden ? false : bookingField.required;
+        const isRequired = bookingField.hidden
+          ? false
+          : isFieldApplicableToCurrentView
```

Comment:
If Field is applicable to current view then only it makes sense to consider it required.